sdk/storage/storage-blob/src/Clients.ts

```diff
@@ -3116,15 +3116,25 @@ export class BlockBlobClient extends BlobClient {
    * ```
    */
   public async upload(
-    body: HttpRequestBody,
+    body: HttpRequestBody | NodeJS.ReadableStream,
     contentLength: number,
     options: BlockBlobUploadOptions = {}
   ): Promise<BlockBlobUploadResponse> {
     options.conditions = options.conditions || {};
     const { span, spanOptions } = createSpan("BlockBlobClient-upload", options.tracingOptions);
     try {
+      let newBody: HttpRequestBody;
+      if (typeof (body as any).pipe === "function") {
+        let factory = function(): NodeJS.ReadableStream {
+          return body as NodeJS.ReadableStream;
+        };
+        Object.defineProperty(factory, "retriable", { value: false });
```

Comment:
```suggestion
        Object.defineProperty(factory, "retryable", { value: false });
```

Reply:
hmm, unfortunately Storage already has "retriable" in current code...